pkg/util/slice_operations.go

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package util
+
+import (
+	"sync"
+)
+
+type idxRange struct {
+	from int
+	to   int
+}
+
+var idxRangesPool = sync.Pool{
```

Comment:
Why do we need a pool for this? Pools are not cheap: they make garbage collection slower, their use requires additional interface conversions.

Reply:
Pool removed in https://github.com/grafana/mimir/pull/2703/commits/12a98bf242b0da30acd287a17cc0fbc2a43b5f59.

Reply:
thx!